src/borg/testsuite/compress.py

```diff
@@ -157,60 +115,47 @@ def test_auto():
-
-    cs = CompressionSpec("obfuscate,6,zstd,3")
-    assert isinstance(cs.inner.compressor, ZSTD)
+@pytest.mark.parametrize(
+    "specs, c_type, result_range, obfuscate_factor",
+    [
+        ("obfuscate,1,none", CNONE, 50, 101),
+        ("obfuscate,110,none", CNONE, None, 1024),
+        ("obfuscate,2,lz4", LZ4, 10, 1001),
+        ("obfuscate,6,zstd,3", ZSTD, 90, 10000001),
+        ("obfuscate,2,auto,zstd,10", Auto, 10, 1001),
+    ],
+)
+def test_obfuscate(specs, c_type, result_range, obfuscate_factor):
+    cs = CompressionSpec(specs)
+    assert isinstance(cs.inner.compressor, c_type)
     compressor = cs.compressor
     data = bytes(10000)
     _, compressed = compressor.compress({}, data)
-    min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-    assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * 10000001
-    # compressing 100 times the same data should give multiple different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 90
 
-    cs = CompressionSpec("obfuscate,2,auto,zstd,10")
-    assert isinstance(cs.inner.compressor, Auto)
-    compressor = cs.compressor
-    data = bytes(10000)
-    _, compressed = compressor.compress({}, data)
-    min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-    assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * 1001
-    # compressing 100 times the same data should give multiple different result sizes
-    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > 10
-
-    cs = CompressionSpec("obfuscate,110,none")
-    assert isinstance(cs.inner.compressor, CNONE)
-    compressor = cs.compressor
-    data = bytes(1000)
-    _, compressed = compressor.compress({}, data)
-    assert 1000 <= len(compressed) <= 1000 + 1024
-    data = bytes(1100)
-    _, compressed = compressor.compress({}, data)
-    assert 1100 <= len(compressed) <= 1100 + 1024
+    # tests obfuscation with compression
+    if not isinstance(cs.inner.compressor, CNONE):
+        min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
+        assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * obfuscate_factor
+        assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > result_range
+        # compressing 100 times the same data should give multiple different result sizes
+
+    # tests obfuscation without compression
+    elif specs == "obfuscate,1,none":
+        assert len(data) <= len(compressed) <= len(data) * obfuscate_factor
+        assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > result_range
+    elif specs == "obfuscate,110,none":
+        data = bytes(1000)
+        _, compressed = compressor.compress({}, data)
+        assert 1000 <= len(compressed) <= 1000 + obfuscate_factor
```

Comment:
it's a bit ugly that the additive obfuscation is now tested with a "factor" (which implies multiplication).

rename it?
split into separate tests for additive and multiplicative obfuscation? maybe also separate the none compression and get rid of the if/elif?

Reply:
This makes a lot of sense, I will separate these and tests.

However, I am still unclear on why obfuscation level 1 gets an obfuscation 'factor' of 101, while obfuscation level 110 gets an obfuscation 'addition' of 1024?

Reply:
it is just defined like that (abusing one levels range for 2 different obfuscators):

https://borgbackup.readthedocs.io/en/2.0.0b5/usage/help.html?highlight=obfuscate#borg-help-compression (you can also read the code)